integrations/jenkins/client.py

```diff
@@ -45,6 +46,36 @@ async def get_builds(self) -> AsyncGenerator[list[dict[str, Any]], None]:
             event.attributes.setdefault(ResourceKey.BUILDS, []).extend(builds)
             yield builds
 
+    async def get_build_stages(self, build_url: str) -> list[dict[str, Any]]:
+        response = await self.client.get(f"{build_url}/wfapi/describe")
+        response.raise_for_status()
+        stages = response.json()["stages"]
+
+        # Enrich stages with full URLs
+        for stage in stages:
+            relative_url = stage["_links"]["self"]["href"]
+            stage["__fullUrl"] = f"{self.jenkins_base_url}{relative_url}"
+
+        return stages
+
+    async def get_stages(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        if cache := event.attributes.get(ResourceKey.STAGES):
```

Comment:
we now have a helper function that handles caching. Maybe we can use that instead of manually doing it ourselves?

from port_ocean.utils.cache import cache_iterator_result

@cache_iterator_result()
async def get_stages():
   do something()